### PR TITLE
Algorithm0r/sound music

### LINKED ISSUE #20: Adding Mario's sound effects and feed-back
This is not critically important to the game, but any good game has both sound-effects and audio feed-back to really nail home the game's polish.

since they would be implemented in more-or-less the same way: this issue includes the sound-effects that come from mario interacting with things or taking actions. this issue is for mario's own actions.

Any additional sound effects that need to be implemented to particular enemys (flying, jumping/falling, attacking, etc) will be added later as their importance is not nearly as great as mario's own.

### PR DESCRIPTION
**Pull request by algorithm0r**
# Pull Request Description

Please fill out the form below. For the checkbox, use [x] to have a checkmark. For example:

- [x] With a checkmark.
- [ ] Without a checkmark.

This should not affect the way whether your pull request is accepted or not. Think of it like a survey and documentation, so the reviewers and future developers can follow your PR easier.

1. Have you included *.gitignore* file? Or manually remove config and environment files (E.g: *node_modules* folder, Eclipse IDE files) from this submission? The answer should be **Yes** in this case.

   - [x] Yes
   - [ ] No

2. Did you fully complete the assigned task(s)?

   - [ ] Fully completed
   - [x] Partially completed

3. Did you introduce a new non-required feature in this PR? This feature is not an assigned task, but it is the one that you feel will make the game better. For example, reproduce a bug used by speed runner in an original version.

   - [ ] Yes
   - [x] No

4. Are you aware of any bug caused by your code that have not been fixed in this PR?

   - [ ] Yes
   - [x] No

5. Are you aware of any **potential bug**? You don't have to fix this, it is just a way to document. You may not face the bug during the development process, but theoretically it can happens in edge cases. For example, memory leak or a player falls out of map.

   - [ ] Yes
   - [x] No

## Unimplented task(s)

- Not all sounds implemented.

## Related issue(s)

- ISSUE: audio

## Added

- Background music for overworld and underworld levels.
- Sound effects for jumping, squishing and bouncing/breaking bricks.

## Changed and Enhanced

- AssetManager now plays sounds.
- Added background music to levels.js.

## Bugs

1. Not-fixed bugs (only answer when you chose **Yes** in question 4):
2. Potention bugs (only answer when you chose **Yes** in question 5):

## Contributors and code author(s)

- algorithm0r

## Final check

Have you fully filled out this document, and add a label **PR: NEED REVIEW** to this pull request? Answer *Yes* (as it should be) means you are good to go! Thanks for the contribution.

- [x] Yes
- [ ] No

### CODE REVIEW

**Comment by hunghvu** [comment]
I will hold back on reviewing this PR because some students are also working on related issues.

**Review by hunghvu — APPROVED** [review]
As of 2/22/2021, no students submit PR related to sound, so I will approve this.

File: assetmanager.js [hunk 1]
```diff
@@ -18,31 +18,104 @@ class AssetManager {
...
+                        that.successCount++;
+                        if (that.isDone()) callback();
+                    });
 
-            img.addEventListener("error", function () {
-                console.log("Error loading " + this.src);
-                that.errorCount++;
-                if (that.isDone()) callback();
-            });
+                    img.addEventListener("error", function () {
```
Comment (BenjaminDeJager) [inline assetmanager.js:37]: I see green blocks here, is that just me?

**Review by BenjaminDeJager — APPROVED** [review]

**Comment by BenjaminDeJager** [comment]
the sounds and music adds a surprising amount to the game...wow...

**Pull request merged**